### PR TITLE
[file-server] Add "from" parameter to browse complementary to "limit"

### PR DESCRIPTION
**Pull request by snu-ceyda**
A simple update to the **browse** functionality within the **file-server** 

- Added an optional **from** URL parameter complementary to the existing **limit** parameter.

Use case -> To efficiently browse folders with many... many... files

## Usage

caddy file-server --listen 8000 --browse

http://localhost:8000/browse
Shows all files

http://localhost:8000/browse?limit=10
Shows only files 10 files starting from 0

http://localhost:8000/browse?limit=10&from=5
Shows only files 10 files starting from 5

### CODE REVIEW

**Comment by mholt** [comment]
Thanks for the PR! I'll try to give it a pass soon.

**Comment by snu-ceyda** [comment]
> I think `offset` would be better than `from`.
that's reasonable too, should I change it?

File: modules/caddyhttp/fileserver/browse.go [hunk 1]
```diff
@@ -106,6 +106,7 @@ func (fsrv *FileServer) browseApplyQueryParams(w http.ResponseWriter, r *http.Re
 	sortParam := r.URL.Query().Get("sort")
 	orderParam := r.URL.Query().Get("order")
 	limitParam := r.URL.Query().Get("limit")
+	fromParam := r.URL.Query().Get("from")
```
Comment (mholt) [inline modules/caddyhttp/fileserver/browse.go:109]: As discussed, change to `offset`.
```suggestion
	offsetParam := r.URL.Query().Get("offset")
```

File: modules/caddyhttp/fileserver/browselisting.go [hunk 2]
```diff
@@ -101,6 +101,9 @@ type browseListing struct {
 
 	// If ≠0 then Items have been limited to that many elements.
 	ItemsLimitedTo int
+
+	// If ≠0 then Items starting from that many elements.
+	ItemsFrom int
```
Comment (mholt) [inline modules/caddyhttp/fileserver/browselisting.go:106]:
```suggestion
	ItemOffset int
```

File: modules/caddyhttp/fileserver/browselisting.go [hunk 3]
```diff
@@ -131,7 +134,7 @@ func (l browseListing) Breadcrumbs() []crumb {
 	return result
 }
 
-func (l *browseListing) applySortAndLimit(sortParam, orderParam, limitParam string) {
+func (l *browseListing) applySortAndLimit(sortParam, orderParam, limitParam string, fromParam string) {
```
Comment (mholt) [inline modules/caddyhttp/fileserver/browselisting.go:137]:
```suggestion
func (l *browseListing) applySortAndLimit(sortParam, orderParam, limitParam string, offsetParam string) {
```

File: modules/caddyhttp/fileserver/browselisting.go (hunk 2 above)
Comment (snu-ceyda) [reply modules/caddyhttp/fileserver/browselisting.go]: Also changed the browse template file to 'offset'.

**Pull request merged**